ejb/src/main/java/org/wildfly/httpclient/ejb/HttpDiscoveryHandler.java

```diff
@@ -70,16 +71,18 @@ public void moduleUnavailable(List<EJBModuleIdentifier> modules) {
     @Override
     protected void handleInternal(HttpServerExchange exchange) throws Exception {
         exchange.getResponseHeaders().put(Headers.CONTENT_TYPE, Constants.EJB_DISCOVERY_RESPONSE.toString());
+        byte[] data;
         final ByteArrayOutputStream out = new ByteArrayOutputStream();
         Marshaller marshaller = httpServiceConfig.getHttpMarshallerFactory(exchange)
                 .createMarshaller(HttpProtocolV1ObjectTable.INSTANCE);
-        marshaller.start(new NoFlushByteOutput(Marshalling.createByteOutput(out)));
-        marshaller.writeInt(availableModules.size());
-        for (EJBModuleIdentifier ejbModuleIdentifier : availableModules) {
-            marshaller.writeObject(ejbModuleIdentifier);
+        ByteOutput byteOutput = byteOutputOf(out);
```

Comment:
Can't this be included as try with resources block (i.e. include the ByteOutput instance as a resource that will be closed when the try block exits?)

Reply:
Fixed as you suggested.